web/src/ee/features/billing/server/stripeWebhookApiHandler.ts

```diff
@@ -0,0 +1,311 @@
+    return NextResponse.json(
+      { message: `Webhook Error: ${err}` },
+      { status: 400 },
+    );
+  }
+
+  // Handle the event
+  switch (event.type) {
+    case "checkout.session.completed":
+      // add customer id and subscription id to the organization that created the checkout session
+      const session = event.data.object;
+      console.log("[Stripe Webhook] Start checkout.session.completed", session);
+      await handleCheckoutSessionCompleted(session);
+      break;
+    case "customer.subscription.created":
+      // update the active product id on the organization linked to the subscription
+      const subscription = event.data.object;
+      console.log(
+        "[Stripe Webhook] Start customer.subscription.created",
+        subscription,
+      );
+      await handleSubscriptionChanged(subscription, "created");
+      break;
+    case "customer.subscription.deleted":
+      // remove the active product id on the organization linked to the subscription
+      const deletedSubscription = event.data.object;
+      console.log(
+        "[Stripe Webhook] Start customer.subscription.deleted",
+        deletedSubscription,
+      );
+      await handleSubscriptionChanged(deletedSubscription, "deleted");
+      break;
+    case "customer.subscription.updated":
+      // update the active product id on the organization linked to the subscription
+      const updatedSubscription = event.data.object;
+      console.log(
+        "[Stripe Webhook] Start customer.subscription.updated",
+        updatedSubscription,
+      );
+      await handleSubscriptionChanged(updatedSubscription, "updated");
+      break;
+    default:
+      console.log(`Unhandled event type ${event.type}`);
+  }
+
+  return NextResponse.json({ received: true }, { status: 200 });
+}
+
+async function handleCheckoutSessionCompleted(
+  session: Stripe.Checkout.Session,
+) {
+  // the client reference is passed to the stripe checkout session via the pricing page
+  const clientReference = session.client_reference_id;
+  if (!clientReference) {
+    console.error("[Stripe Webhook] No client reference");
+    return NextResponse.json(
+      { message: "No client reference" },
+      { status: 400 },
+    );
+  }
+  if (!isStripeClientReferenceFromCurrentCloudRegion(clientReference)) {
+    console.error(
+      "[Stripe Webhook] Client reference not from current cloud region",
+    );
+    return;
+  }
+
+  const orgId = getOrgIdFromStripeClientReference(clientReference);
+  const subscriptionId = session.subscription; // Assuming subscription ID is in session.subscription
+
+  // check if org exists and that the customer and subscription IDs match
+  const organization = await prisma.organization.findUnique({
+    where: { id: orgId },
+  });
+  if (!organization) {
+    console.error("[Stripe Webhook] Organization not found");
+    Sentry.captureMessage("[Stripe Webhook] Organization not found", {
```

Comment:
Please use the otel integration now

`traceException` from `import { traceException } from "@langfuse/shared/src/server";`